pkg/network/ingress/ingress.go

```diff
@@ -68,12 +68,12 @@ func InsertProbe(ing *v1alpha1.Ingress) (string, error) {
 // HostsPerVisibility takes an Ingress and a map from visibility levels to a set of string keys,
 // it then returns a map from that key space to the hosts under that visibility.
 func HostsPerVisibility(ing *v1alpha1.Ingress, visibilityToKey map[v1alpha1.IngressVisibility]sets.String) map[string]sets.String {
-	output := make(map[string]sets.String)
+	output := make(map[string]sets.String, 2) // We currently have public and internal.
 	for _, rule := range ing.Spec.Rules {
 		for host := range ExpandedHosts(sets.NewString(rule.Hosts...)) {
 			for key := range visibilityToKey[rule.Visibility] {
 				if _, ok := output[key]; !ok {
-					output[key] = sets.NewString()
+					output[key] = make(sets.String, 1)
```

Comment:
this seems like it should be:
 
```suggestion
					output[key] = make(sets.String, len(visibilityToKey[rule.Visibility]))
```

Reply:
oops, no, I mean

```suggestion
					output[key] = make(sets.String, len(ExpandedHosts(sets.NewString(rule.Hosts...)))
```

Reply:
So we're allocating an array to determine how many items we should preallocate? 🤔

Reply:
haha yeah, Im not sure this one is worth the bother (but I'm pretty sure 1 isn't the right number, better to just stick with map defaults than that)

Reply:
yeah, I think 2 is more correct... but :shrug: one less allocation.

Reply:
we can go with len(hosts) to be preventive. I can get behind that.